src/core/namespace.c

```diff
@@ -1618,6 +1620,11 @@ static int apply_one_mount(
                  * that bind mount source paths are always relative to the host root, hence we pass NULL as
                  * root directory to chase() here. */
 
+                /* When we create implicit mounts, we might need to create the path ourselves as it is on a
+                 * just-created tmpfs, for example. */
+                if (m->create_source_dir)
+                        (void) mkdir_p(mount_entry_path(m), m->source_dir_mode);
```

Comment:
mkdir_p is already quite defensive, hence I'd really just catch errors here and write a friendly debug message